splink/duckdb/duckdb_linker.py

```diff
@@ -170,29 +166,23 @@ def _table_to_splink_dataframe(
         return DuckDBLinkerDataFrame(templated_name, physical_name, self)
 
     def _execute_sql_against_backend(self, sql, templated_name, physical_name):
-
         # In the case of a table already existing in the database,
         # execute sql is only reached if the user has explicitly turned off the cache
         self._delete_table_from_database(physical_name)
 
-        logger.debug(
-            execute_sql_logging_message_info(
-                templated_name, self._prepend_schema_to_table_name(physical_name)
-            )
-        )
-        logger.log(5, log_sql(sql))
-
         sql = f"""
         CREATE TABLE {physical_name}
         AS
         ({sql})
         """
-        self._con.execute(sql).fetch_df()
+        self._log_and_do_execute_sql(sql, templated_name, physical_name)
 
         return DuckDBLinkerDataFrame(templated_name, physical_name, self)
 
-    def register_table(self, input, table_name, overwrite=False):
+    def _do_execute_sql(self, final_sql, templated_name, physical_name):
+        self._con.execute(final_sql).fetch_df()
```

Comment:
wait, we can just remove the .fetch_df() call, right?

Reply:
removed in follow up commit

Reply:
Yeah, good spot. It doesn't actually materialise the whole df, it just reports back the total row count for the materialised df.

Either way, it's a legacy bit of code from several months ago that we can remove.